### ISSUE TITLE
time-1.8 (and master) fails to build on i386

### ISSUE THREAD

**Issue by trofi** [haskell/time#65]
`CTime` there is `Int32`.
```
[21 of 33] Compiling Data.Time.LocalTime.Internal.TimeZone ( lib/Data/Time/LocalTime/Internal/TimeZone.hs, dist/build/Data/Time/LocalTime/Internal/TimeZone.o )

lib/Data/Time/LocalTime/Internal/TimeZone.hs:101:48: error:
    • Couldn't match type ‘Int64’ with ‘Int32’
      Expected type: SystemTime -> Int32
        Actual type: SystemTime -> Int64
    • In the second argument of ‘(.)’, namely ‘systemSeconds’
      In the second argument of ‘(.)’, namely ‘CTime . systemSeconds’
      In the expression: getTimeZoneCTime . CTime . systemSeconds

```

**Comment by AshleyYakeley** [comment]
This is awkward, because even on modern 32-bit systems, time_t is 32 bits, leading to the [Year 2038 problem](https://en.wikipedia.org/wiki/Year_2038_problem).

**Comment by AshleyYakeley** [comment]
Annoyingly, there's no `instance Bounded CTime`...

**Comment by AshleyYakeley** [comment]
This is now fixed in master, and time builds on my 32 bit Ubuntu VM. However, lots of QuickCheck tests fail -- this may be unavoidable.

**Comment by AshleyYakeley** [comment]
@bgamari: this fix is in 1.8.0.1, which has been released. I've updated the "ghc" branch, so you can update your submodule.
